langchain/document_loaders/utils/metadata.py

```diff
@@ -0,0 +1,83 @@
+"""Interface for retrieving metadata from the various document loaders."""
+import os
+import platform
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import IO, Optional
+
+from langchain.document_loaders.utils.filetype import get_mime_type, get_extension
+
+
+class BaseMetadata(ABC):
+    """Sample metadata to be extracted from the different types of loaded documents."""
+
+    @abstractmethod
```

Comment:
Leaving these here as abstract methods because we still don't know what are all the potential methods we have in mind. Future contributors can also refer here for implementation guide or even ideas for new methods.